### ISSUE TITLE
Is there any memory-efficient tensorflow Implementation？

### ISSUE THREAD

**Issue by feynman233** [liuzhuang13/DenseNet#32]
Hi, I have found the same problem about the GPU memory，and read your report（https://arxiv.org/pdf/1707.06990.pdf）.
So, is there any memory-efficient tensorflow Implementation？
Thanks very much!

**Comment by liuzhuang13** [comment]
Sorry we don't have any memory-efficient tensorflow implementation, as far as I know.

But I think for practical usage, you don't always need a memory-efficient implementation. Note that memory-efficient version still costs you at least the same amout of time. In most of time, if you have a 4-GPU machine, naive implementation is enough. Thanks!

**Comment by joeyearsley** [comment]
https://github.com/joeyearsley/efficient_densenet_tensorflow

Made using gradient checkpointing like the gpleiss repo does.
